### PR TITLE
Add validation to ci

### PR DESCRIPTION
**Pull request by aurelien-reeves**
### 🤔 What's changed?

In the monorepo, there was some acceptance testing in the makefile.
Those acceptance tests are minimalistic. They have  been moved as part of the mocha test suite, but in a distinct npm script: "validate"

Now to validate that fake-cucumber can process feature files without errors, run `npm run validate`

It has also been added to the github workflow

### ⚡️ What's your motivation? 

Validate that fake-cucumber can process feature file without errors as part of the CI without the makefile issued from the monorepo

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by mattwynne** [comment]
It would probably be helpful to document this somewhere. I notice there's currently no `CONTRIBUTING.md` guide at all, so perhaps that's a problem for another PR.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -39,6 +39,6 @@ jobs:
 
       - run: npm install-ci-test
 
-      - run: npm run build
+      - run: npm run validate
```
Comment (mattwynne) [inline .github/workflows/test.yml:42]: I find `validate` a bit unclear.

what about naming the two jobs `npm run test:unit` and `npm run test:acceptance`?

Then we can have an `npm run test` that runs both of them.

Comment (mattwynne) [inline .github/workflows/test.yml:42]:
```suggestion
      - run: npm test
```

File: package.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
     "fake-cucumber": "bin/fake-cucumber"
   },
   "scripts": {
-    "test": "mocha",
+    "test": "mocha --exclude test/acceptance.ts",
```
Comment (mattwynne) [inline package.json:11]:
```suggestion
    "test": "npm run test:unit && npm run test:acceptance",
    "test:unit": "mocha --exclude test/acceptance.ts",
```

File: package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
...
+    "validate": "npm run build && mocha test/acceptance.ts",
```
Comment (mattwynne) [inline package.json:12]:
```suggestion
    "test:acceptance": "npm run build && mocha test/acceptance.ts",
```

**Review by mattwynne — APPROVED** [review]
Suggestions for you to consider.

**Review by davidjgoss — APPROVED** [review]

**Pull request merged**